### PR TITLE
[flow-strict] Flow TouchableNativeFeedback.android.js

### PR DESCRIPTION
**Pull request by mottox2**
Related to #22100

Turn Flow strict mode on for Libraries/Components/Touchable/TouchableNativeFeedback.android.js.

Test Plan:
----------
- [x] npm run prettier
- [x] npm run flow
- [x] npm run flow-check-ios
- [x] npm run flow-check-android
- [x] npm run lint
- [x] npm run test
 
Changelog:
----------
[GENERAL][FIXED] - apply flow

### CODE REVIEW

**Comment by mottox2** [comment]
Sorry. I will submit PR to apply flow strict to `Touchable.js` before this PR.

File: Libraries/Components/Touchable/TouchableNativeFeedback.android.js [hunk 1]
```diff
@@ -13,6 +14,7 @@ const Platform = require('Platform');
 const React = require('React');
 const PropTypes = require('prop-types');
 const ReactNative = require('ReactNative');
+// $FlowFixMe
```
Comment (TheSavior) [inline Libraries/Components/Touchable/TouchableNativeFeedback.android.js]: This is to fix the flow warning about not importing from untyped files, right? If that is the case then lets just not turn on strict-local in this file.

Comment (mottox2) [reply Libraries/Components/Touchable/TouchableNativeFeedback.android.js]: Thank you for reviewing.
I applied flow to `Touchable.js` then I removed `FlowFixMe` from `TouchableNativeFeedback.android.js`.

File: Libraries/Components/Touchable/Touchable.js [hunk 2]
```diff
@@ -652,12 +698,12 @@ const TouchableMixin = {
     );
   },
 
-  _handleDelay: function(e) {
+  _handleDelay: function(e: LayoutEvent) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: This is a bit curious. How come you decided to type this with `LayoutEvent`?

File: Libraries/Components/Touchable/Touchable.js [hunk 3]
```diff
@@ -652,12 +698,12 @@ const TouchableMixin = {
...
     this.touchableDelayTimeout = null;
     this._receiveSignal(Signals.DELAY, e);
   },
 
-  _handleLongDelay: function(e) {
+  _handleLongDelay: function(e: LayoutEvent) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: Same deal here. Why is this a `LayoutEvent` and not a `PressEvent`? This function calls `_receiveSignal`, which I think should accept a `PressEvent`.

File: Libraries/Components/Touchable/Touchable.js [hunk 4]
```diff
@@ -725,14 +771,14 @@ const TouchableMixin = {
...
-  _isHighlight: function(state) {
+  _isHighlight: function(state: State) {
     return (
       state === States.RESPONDER_ACTIVE_PRESS_IN ||
       state === States.RESPONDER_ACTIVE_LONG_PRESS_IN
     );
   },
 
-  _savePressInLocation: function(e) {
+  _savePressInLocation: function(e: LayoutEvent) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: Later on in the function, we call `TouchEventUtils.extractSingleTouch`, which expects to be called with the native event of a `PressEvent`. So, I think this type should be `PressEvent` and not `LayoutEvent`.

File: Libraries/Components/Touchable/Touchable.js [hunk 5]
```diff
@@ -813,12 +869,12 @@ const TouchableMixin = {
...
   },
 
-  _startHighlight: function(e) {
+  _startHighlight: function(e: LayoutEvent) {
     this._savePressInLocation(e);
     this.touchableHandleActivePressIn && this.touchableHandleActivePressIn(e);
   },
 
-  _endHighlight: function(e) {
+  _endHighlight: function(e: LayoutEvent) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: This method calls `this.touchableHandleActivePressOut`, which you typed as accepting a `PressEvent` inside `TouchableNativeFeedback.android.js`. So, shouldn't the type of `e` also be `PressEvent`?

File: Libraries/Components/Touchable/Touchable.js [hunk 6]
```diff
@@ -813,12 +869,12 @@ const TouchableMixin = {
     UIManager.playTouchSound();
   },
 
-  _startHighlight: function(e) {
+  _startHighlight: function(e: LayoutEvent) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: Same deal as `_endHighlight` here.

File: Libraries/Components/Touchable/Touchable.js [hunk 7]
```diff
@@ -758,7 +809,12 @@ const TouchableMixin = {
    * @param {Event} e Native event.
    * @sideeffects
    */
-  _performSideEffectsForTransition: function(curState, nextState, signal, e) {
+  _performSideEffectsForTransition: function(
+    curState: State,
+    nextState: State,
+    signal: Signal,
+    e: LayoutEvent,
+  ) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: This event is forwarded to `_startHighlight` and `_endHighlight`. Shouldn't it therefore also be typed as a `PressEvent`?

File: Libraries/Components/Touchable/Touchable.js [hunk 8]
```diff
@@ -685,7 +731,7 @@ const TouchableMixin = {
    * @throws Error if invalid state transition or unrecognized signal.
    * @sideeffects
    */
-  _receiveSignal: function(signal, e) {
+  _receiveSignal: function(signal: Signal, e: LayoutEvent) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: This gets forwarded to `_performSideEffectsForTransition`. So, shouldn't `e` be typed as `PressEvent`?

File: Libraries/Components/Touchable/Touchable.js [hunk 9]
```diff
@@ -391,7 +430,7 @@ const TouchableMixin = {
    * @param {SyntheticEvent} e Synthetic event from event system.
    *
    */
-  touchableHandleResponderGrant: function(e) {
+  touchableHandleResponderGrant: function(e: SyntheticEvent<any>) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: I think this function accepts a PressEvent.

File: Libraries/Components/Touchable/Touchable.js [hunk 10]
```diff
@@ -432,21 +471,21 @@ const TouchableMixin = {
...
-  touchableHandleResponderRelease: function(e) {
+  touchableHandleResponderRelease: function(e: PressEvent) {
     this._receiveSignal(Signals.RESPONDER_RELEASE, e);
   },
 
   /**
    * Place as callback for a DOM element's `onResponderTerminate` event.
    */
-  touchableHandleResponderTerminate: function(e) {
+  touchableHandleResponderTerminate: function(e: PressEvent) {
```
Comment (RSNara) [inline Libraries/Components/Touchable/Touchable.js]: You've typed this function as accepting a `PressEvent`, but later on in the function, we forward this event to `_receiveSignal`, which accepts a LayoutEvent. There's a contradiction here.

**Pull request closed without merging**